internal/validate/validate_test.go

```diff
@@ -241,6 +246,385 @@ func TestValidate(t *testing.T) {
+		},
+		{
+			failurePolicy:              admissionregistration.Fail,
+			paramNotFoundAction:        admissionregistration.DenyAction,
+			invalidResource:            true,
+			paramsFound:                false,
+			expectedValidationResponse: rejectionResponseParamsNotFound,
+		},
+		{
+			failurePolicy:              admissionregistration.Fail,
+			paramNotFoundAction:        admissionregistration.DenyAction,
+			invalidResource:            false,
+			paramsFound:                false,
+			expectedValidationResponse: rejectionResponseParamsNotFound,
+		},
+		{
+			failurePolicy:              admissionregistration.Ignore,
+			paramNotFoundAction:        admissionregistration.DenyAction,
+			invalidResource:            true,
+			paramsFound:                true,
+			expectedValidationResponse: rejectionResponse,
+		},
+		{
+			failurePolicy:              admissionregistration.Ignore,
+			paramNotFoundAction:        admissionregistration.DenyAction,
+			invalidResource:            false,
+			paramsFound:                true,
+			expectedValidationResponse: acceptanceResponse,
+		},
+		{
+			failurePolicy:              admissionregistration.Ignore,
+			paramNotFoundAction:        admissionregistration.DenyAction,
+			invalidResource:            true,
+			paramsFound:                false,
+			expectedValidationResponse: acceptanceResponse,
+		},
+		{
+			failurePolicy:              admissionregistration.Ignore,
+			paramNotFoundAction:        admissionregistration.DenyAction,
+			invalidResource:            false,
+			paramsFound:                false,
+			expectedValidationResponse: acceptanceResponse,
+		},
+		{
+			failurePolicy:              admissionregistration.Fail,
+			paramNotFoundAction:        admissionregistration.AllowAction,
+			invalidResource:            true,
+			paramsFound:                true,
+			expectedValidationResponse: rejectionResponse,
+		},
+		{
+			failurePolicy:              admissionregistration.Fail,
+			paramNotFoundAction:        admissionregistration.AllowAction,
+			invalidResource:            false,
+			paramsFound:                true,
+			expectedValidationResponse: acceptanceResponse,
+		},
+		{
+			failurePolicy:              admissionregistration.Fail,
+			paramNotFoundAction:        admissionregistration.AllowAction,
+			invalidResource:            true,
+			paramsFound:                false,
+			expectedValidationResponse: acceptanceResponse,
+		},
+		{
+			failurePolicy:              admissionregistration.Fail,
+			paramNotFoundAction:        admissionregistration.AllowAction,
+			invalidResource:            false,
+			paramsFound:                false,
+			expectedValidationResponse: acceptanceResponse,
+		},
+		{
+			failurePolicy:              admissionregistration.Ignore,
+			paramNotFoundAction:        admissionregistration.AllowAction,
+			invalidResource:            true,
+			paramsFound:                true,
+			expectedValidationResponse: rejectionResponse,
```

Comment:
Shouldn't this one be acceptanceResponse, since the resource is invalid and the failurePolicy is Ignore?

(In VAPs "The absence of matched parameters is subject to the failurePolicy of the policy")